File: dart/elevation_peak/lib/data_config.dart
import 'dart:io';

const path = 'data/data.txt';

class DataConfig {
  var rows = 0;
  var cols = 0;

  var minPeak = 0;
  var exRadius = 0;

  late final List<List<int>> data;

  DataConfig() {
    read(path);
  }

  void read(String path) {
    try {
      final input = File.fromUri(Uri.file(path)).readAsLinesSync().iterator;
      if (input.moveNext()) {
        final line = input.current.split(RegExp(r'\s'));
        rows = int.parse(line[0]); // Number of rows in dataset.
        cols = int.parse(line[1]); // Number of columns in dataset.
        minPeak = int.parse(line[2]); // Minimum peak value.
        exRadius = int.parse(line[3]); // The exclusion radius.
      }
      // Create array of size [ROWS] * [COLS].
      data = List.generate(rows, (_) => List.generate(cols, (_) => 0));
      for (var i = 0; i < rows && input.moveNext(); i++) {
        final elements = input.current.split(RegExp(r'\s')).iterator;
        for (var j = 0; j < cols && elements.moveNext(); j++) {
          data[i][j] = int.parse(elements.current);
        }
      }
    } on FileSystemException {
      print('File does not exists.');
    }
  }
}

File: kotlin/Elevation/src/DataConfig.kt
import java.io.File
import java.io.FileNotFoundException
import java.util.*

class DataConfig {
    private companion object {
        private const val PATH = "data/data.txt"
    }

    // Size of the dataset
    var rows = 0; private set
    var cols = 0; private set

    // Current minimum peak
    var minPeak = 0; private set

    // Exclusion radius
    var exRadius = 0; private set

    var data: Array<IntArray> = arrayOf(); private set

    init {
        read(PATH)
    }

    /**
     * Read data from the "data.txt" file.
     *
     * @return two-dimensional integer array with data.
     */
    private fun read(path: String) {
        try {
            Scanner(File(path)).let { input ->
                if (input.hasNextLine()) {
                    rows = input.nextInt() // Number of rows in dataset.
                    cols = input.nextInt() // Number of columns in dataset.
                    minPeak = input.nextInt() // Minimum peak value.
                    exRadius = input.nextInt() // The exclusion radius.
                }
                // Create array of size [ROWS] * [COLS].
                data = Array(rows) { IntArray(cols) }
                repeat(rows) { i ->
                    repeat(cols) { j -> data[i][j] = input.nextInt() }
                }
            }
        } catch (e: FileNotFoundException) {
            println("File does not exists.")
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}
